Forte.ContentfulSchema.Tests/Core/ContentTypeDefinitionTests.cs

```diff
@@ -159,31 +131,69 @@ public void ShouldReturnFalseWhenUpdatingSameOrIncorrectEditorInterfaces(List<Ed
-            
-            return editor;
-        }
+           new Action<List<Field>>[] { f =>
+                {
+                    f.Clear();
+                    f.Add(new Field {Id = "NewField1"});
+                }
+            },
+            new Action<List<Field>>[] { f => f[0].Name = "ChangedFieldName" },
+            new Action<List<Field>>[] { f => f[0].Type = "Text" },
+            new Action<List<Field>>[] { f => f[0].Omitted = true },
+            new Action<List<Field>>[] { f => f[0].Localized = true },
+            new Action<List<Field>>[] { f => f[0].Required = true },
+            new Action<List<Field>>[] { f => f[0].LinkType = "Entry" },
+            new Action<List<Field>>[] { f => f[0].Disabled = true },
+            new Action<List<Field>>[] { f => f[0].Validations = new List<IFieldValidator>(){new UniqueValidator()} },
+            new Action<List<Field>>[] { f => f[0].Items = new Schema(){LinkType = "TestLinkType"}},
+            new Action<List<Field>>[] { f => f[0].Items = new Schema(){Type = "TestType"}},
+            new Action<List<Field>>[] { f => f[0].Items = new Schema(){Validations = new List<IFieldValidator>(){new UniqueValidator()}}},
+            new Action<List<Field>>[] { f => f.Clear() },
+            new Action<List<Field>>[] { f => f.AddRange(new List<Field>{ new Field {Id = "NewField1"}, new Field {Id = "NewField2"} }) },
+        };
 
-//        [Theory]
-//        [InlineData(false)]
-//        [InlineData(true)]
-//        public void ShouldReturnValueDependingFromTheFieldComparerResult(bool areEqual)
-//        {
-//            var fieldComparer = new Mock<IEqualityComparer<Field>>();
-//            fieldComparer.Setup(m => m.Equals(It.IsAny<Field>(), It.IsAny<Field>()))
-//                .Returns(areEqual);
+        public static readonly Action<List<Field>> OrderFieldsModificators = f =>
+        {
+            f.Clear();
+            f.Add(new Field {Id = "NewField1"});
+            f.Add(new Field {Id = "NewField2"});
+        };
 
-//            var customComparer = new ContentTypeComparer(fieldComparer.Object);
+        public static Action<List<Field>> OrderFieldsModificatorsReverse = f => { f.Reverse(); };
 
-//            var firstContentType = ContentTypeBuilder.New.WithFields(new Field {Id = "1"}).Build();
-//            var secondContentType = ContentTypeBuilder.New.WithFields(new Field {Id = "1"}).Build();
+        private static ContentType CreateDefaultContentType()
+        {
+            return new ContentType
+            {    
+                SystemProperties = new SystemProperties{Id = "123"},
+                Description = "Description",
+                DisplayField = "DisplayField",
+                Name = "Name",
+                Fields = new List<Field>{ new Field{ Id = "field1"} }
```

Comment:
Add one more fields so we can test changing order as one of the modification cases